fix: Null pointer exception when pubKeyHash is null

sitaram-kalluri opened the pull request:
<!--
Please make sure you've read and understood our contributing guidelines in CONTRIBUTING.md

If this is a bug fix, make sure your description includes "fixes #xxxx", or
"closes #xxxx"

Please provide the following information:
-->

**- What I did**
- Fixes https://github.com/atsign-foundation/at_libraries/issues/521

**- How I did it**
- when `fullJson` is set to true (the default value), the resultant map contains all the fields. If the value is not set, null is populated. To be consistent with the other fields, changed the implementation as `pubKeyHash!.toJson()` to `pubKeyHash?.toJson()`

**- How to verify it**
- Raised a draft PR in at_client_sdk to verify the changes: https://github.com/atsign-foundation/at_client_sdk/pull/1245
- Added unit tests to assert the changes.

**- Description for the changelog**
- fix: Null pointer exception when pubKeyHash is null
<!--
Write a short (one line) summary that describes the changes in this
pull request for inclusion in the changelog:
-->

sitaram-kalluri committed:
fix: Null pointer exception when pubKeyHash is null

gkc reviewed (changes requested):
Please add unit tests which would have caught this issue had they existed.

sitaram-kalluri committed:
fix: Add unit tests for metadata

sitaram-kalluri commented:
> Please add unit tests which would have caught this issue had they existed.

Added unit tests and also changed the  approach in fix the bug.

when fullJson is set to true (the default value), the resultant map contains all the fields. If the value is not set, null is populated. To be consistent with the other fields, changed the implementation as `pubKeyHash!.toJson()` to `pubKeyHash?.toJson()`

gkc commented on packages/at_commons/test/at_key_test.dart:
```diff
@@ -987,4 +987,60 @@ void main() {
+      ..isCached = true
+      ..sharedKeyEnc = 'dummy_shared_key_enc'
+      ..pubKeyHash = PublicKeyHash('test_hash', PublicKeyHashingAlgo.sha256)
+      ..isBinary = true;
```
Please extend to set all fields on the metadata and expect on all of them later

sitaram-kalluri replied:
The change is implemented in https://github.com/atsign-foundation/at_libraries/pull/523/commits/8f9f23f468c445106d5f94347e2b42477b9e17f8

sitaram-kalluri committed:
Merge branch 'trunk' into fix_pubKeyHash_bug

sitaram-kalluri committed:
fix: Add fields to the existing unit tests
